### ISSUE TITLE
Elementor container handles disappear when using Post Content

### ISSUE THREAD

**Issue by mapsmarketing** [elementor/elementor#22092]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Please see the following video to better illustrate the issue: https://www.loom.com/share/2cabd4a338b643b78ef46dfe73054722

After going through the steps to reproduce the issue, you'll notice when hovering over the container the handles are not showing for the pages you are editing (this excludes the 'Single Page' template itself).

This issue persists even with all other plugins turned off.

### Steps to reproduce

1. Create a 'Single Page' template
2. Add a 'container' to the single page template
3. Inside the container add a 'Post Content' widget
4. Save the template to be used for all pages
5. Edit an existing page
6. Create container on this page

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: MySQL Community Server (GPL) v5.7.41
	PHP Version: 7.4.33
	PHP Memory Limit: 256M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 64M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.2
	Site URL: https://example.com.au/
	Home URL: https://example.com.au/
	WP Multisite: No
	Max Upload Size: 64 MB
	Memory limit: 40M
	Max Memory limit: 512M
	Permalink Structure: /%postname%/
	Language: en-AU
	Timezone: Australia/Melbourne
	Admin Email: serverlogs@mapsmarketing.com.au
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor Child
	Version: 2.4.1
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 2.7.1
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: en_AU
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36

== Active Plugins ==
	Admin Columns Pro
		Version: 4.2.9
		Author: Admin Columns

	Admin Columns Pro - Advanced Custom Fields (ACF)
		Version: 2.2.3
		Author: Admin Columns

	Advanced Custom Fields PRO
		Version: 6.1.4
		Author: WP Engine

	Advanced Editor Tools
		Version: 5.9.0
		Author: Automattic

	Advanced Sidebar Menu
		Version: 9.0.6
		Author: OnPoint Plugins

	All-in-One Event Calendar by Time.ly
		Version: 3.0.0
		Author: Time.ly Network Inc.

	BackupBuddy
		Version: 8.7.4.1
		Author: iThemes

	Better Search Replace
		Version: 1.4.2
		Author: WP Engine

	Classic Editor
		Version: 1.6.3
		Author: WordPress Contributors

	Easy WP SMTP
		Version: 2.1.0
		Author: Easy WP SMTP

	Elementor
		Version: 3.12.1
		Author: Elementor.com

	Elementor Pro
		Version: 3.12.2
		Author: Elementor.com

	Gravity Forms
		Version: 2.7.4
		Author: Gravity Forms

	InfiniteWP - Client
		Version: 1.11.1
		Author: Revmakx

	MailPoet
		Version: 4.11.1
		Author: MailPoet

	One Click Accessibility
		Version: 2.1.0
		Author: One Click Accessibility

	Prime Slider
		Version: 3.6.4
		Author: BdThemes

	Regenerate Thumbnails
		Version: 3.1.5
		Author: Alex Mills (Viper007Bond)

	Revision Control
		Version: 2.3.2
		Author: Dion Hulse

	Simple Custom Post Order
		Version: 2.5.6
		Author: Colorlib

	Video User Manuals
		Version: 2.5.12
		Author: Video User Manuals Pty Ltd

	WCAG 2.0 form fields for Gravity Forms
		Version: 1.7.2
		Author: Adrian Gordon

	White Label CMS
		Version: 2.5
		Author: www.videousermanuals.com

	WP Rollback
		Version: 1.7.3
		Author: GiveWP.com

	Yoast SEO
		Version: 20.5
		Author: Team Yoast

== Must-Use Plugins ==
	InfiniteWP - Client Loader
		Version: 1.0.1
		Author: Revmakx

== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	

== Elementor Experiments ==
	Optimised DOM Output: Inactive by default
	Improved Asset Loading: Inactive by default
	Improved CSS Loading: Inactive by default
	Inline Font Icons: Inactive by default
	Accessibility Improvements: Active by default
	Additional Custom Breakpoints: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active
	Upgrade Swiper Library: Inactive by default
	Default to New Theme Builder: Inactive
	Hello Theme Header & Footer: Inactive by default
	Editor Top Bar: Inactive by default
	Landing Pages: Active by default
	Nested Elements: Active
	Lazy Load Background Images: Inactive by default
	Page Transitions: Active by default
	Notes: Active by default
	Loop: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default
	Menu: Inactive by default

== Log ==
	
Log: showing 20 of 382023-03-30 11:04:09 [info] Elementor/Upgrades - _on_each_version Start  
2023-03-30 11:04:09 [info] Elementor/Upgrades - _on_each_version Finished 
2023-03-30 11:04:09 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.5',
  'to' => '3.12.0',
)]
2023-03-30 11:04:11 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.11.5',
  'to' => '3.12.0',
)]
2023-04-03 15:03:45 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.7',
  'to' => '3.12.1',
)]
2023-04-03 15:03:45 [info] elementor-pro::elementor_pro_updater Started 
2023-04-03 15:03:45 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-04-03 15:03:45 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-04-03 15:03:45 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.11.7',
  'to' => '3.12.1',
)]
2023-04-11 10:00:21 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.12.1',
  'to' => '3.12.2',
)]
2023-04-11 10:00:21 [info] elementor-pro::elementor_pro_updater Started 
2023-04-11 10:00:21 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-04-11 10:00:22 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-04-11 10:00:22 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.12.1',
  'to' => '3.12.2',
)]
2023-04-11 10:00:24 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.12.1',
  'to' => '3.12.2',
)]
2023-04-14 13:15:48 [info] elementor::elementor_updater Started 
2023-04-14 13:15:48 [info] Elementor/Upgrades - _on_each_version Start  
2023-04-14 13:15:48 [info] Elementor/Upgrades - _on_each_version Finished 
2023-04-14 13:15:48 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.12.0',
  'to' => '3.12.1',
)]
2023-04-14 13:15:49 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.12.0',
  'to' => '3.12.1',
)]

JS: showing 3 of 3JS: 2023-04-17 00:38:40 [error X 13][https://example.com.au//wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.12.1:3:785745] Cannot read properties of undefined (reading 'localized') 
JS: 2023-04-17 04:19:54 [error X 1][https://example.com.au//wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.12.1:3:594206] Cannot read properties of null (reading 'id') 
JS: 2023-04-17 04:24:03 [error X 1][https://example.com.au//wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.12.1:3:785238] Cannot read properties of undefined (reading 'toLowerCase') 

PHP: showing 2 of 2PHP: 2023-04-17 14:17:51 [notice X 1][/home/example/web/wp-content/plugins/elementor/core/common/modules/ajax/module.php::175] Undefined index: data [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-04-18 10:02:38 [notice X 1][/home/example/web/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::168] Undefined index: editor_post_id [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatibility not specified

== Elementor Pro - Compatibility Tag ==

**Comment by mapsmarketing** [comment]
In addition to the above, it also doesn't maintain the set "Gap between elements" from the single page template that was set.

**Comment by mapsmarketing** [comment]
Hi @nicholaszein ,

Just wanted to check if there's been any progress on this bug?

Thanks mate

**Comment by rf-999** [comment]
looking forward for a solution, as this bug is pretty annoying while working with Flexbox/Grid Containers. 
As an Elementor editor I have to create new sections at the bottom of the page as a workaround. On long pages, this is a hassle. 

Hope you can solve this bug soon.

**Comment by nicholaszein** [comment]
Hello, @mapsmarketing and @rf-999! 

Thank you for submitting your issue, and I apologize for the delay! 🙏 

I usually give an answer right when I triage an issue, but I guess this one was missing one. Sorry about that! 😕 

Let me clarify what is happening, and what are our plans for this:

### What is happening
The Elementor Editor is considering **Parent Containers** as **Child Containers**, showing the **editing handle** in the **<ins>top-left</ins>** corner instead of in the **top-center** edge of the current container. From the logic used, it thinks these are child Containers because there is a Parent Container wrapping the Post Content widget in the Theme Builder's Single Template. However...

### This issue does not break functionality
It's important to note that this issue does not break functionality. You can still edit the parent Containers inside your pages. They just don't show the handle in the expected way (at the top-center edge). However, you can still click to edit the Containers, manage them, move them, etc. The only difference is that the handle is located in the top-left corner. You can also still use the Structure Panel to select your Parent Containers, or right-click inside them and select "Edit Container". Please check the video below:


https://github.com/elementor/elementor/assets/113993460/66c44bb1-d1c0-405e-9d3d-763120f53d7f


We understand this might be confusing and not the expected or ideal behavior. So what are we going to do about it?

### Our plans to solve this issue
As you may know, we just released a significant milestone for Elementor Containers: Since version 3.16 Containers will now be active by default on new installations. We want our users to have the best experience they can with the Containers, so we surely need to ensure the User Interface is consistent and intuitive. What you are experiencing here, however, does not fit these conditions. So we will work on fixing this mishap.

### Why haven't we fixed this yet?
In short, because despite it being annoying, it doesn't break the functionality, and we had to prioritize other issues with more pressing matters. We know this is not ideal or something you would want to hear, and we apologize for the inconvenience this might have caused. However, we haven't dismissed this issue, and we do have plans to fix it in a future version. When that happens we will let you know! We can’t say exactly when it'll be sorted out, but we kindly ask for your patience. Thanks for understanding! 😉

Thank you so much for contributing and finding this annoying little egg. 🙏 

I hope this message finds you well.

Kind regards

**Comment by mapsmarketing** [comment]
Hi @nicholaszein ,

Much appreciated!

It's not a breaking issue, but it does confuse clients quite a lot when they are trying to make changes to containers.

Thank you very much

**Comment by tamelander** [comment]
This in very annoying as it makes the theme builder custom page template with containers + content area containers inconsistent behavior. I have switched back to Astra theme default page since this error makes it impossible, when editing a page, to place a new container in the topmost level of the hierarchy between two existing containers. You either have to duplicate an existing container, clear it of content and use it. Or add a container at the bottom and drag it in the navigator.

The conditional code that identifies a topmost container in the hierarchy when editing a page NOT assigned an Elementor custom page template, adds the container handle. The same container when editing the same page assigned an Elementor custom page template does NOT show the handle.  So the functionality is already there, it just need some tweaking to accept a topmost container assigned an Elementor theme page template, as a topmost container and not a child container.

This should get a higher priority since you are now with the latest versions pushing hard for people to start using the container instead of sections. The container and grid learning curve can be steep for many. They don't need containers with inconsistent behavior as a bonus.

**Comment by mocana** [comment]
Any updates on this? I appreciate the thoughts around it not being a priority but i feel it's still a valuable issue to resolve. Having to navigate to the bottom of every page when needing to insert a container at the top and then require you to use the Navigator panel is cumbersome, and does feel broken when you have the feature already established without the use of a template Wrapper.

**Comment by neylwalecki** [comment]
+1

**Comment by ScottMadore** [comment]
Hello, please can you fix this. We need this to work correctly.

### Why it does break functionality
This does in fact break functionality. I can explain very clearly.
Just go try and add a new section in the middle of your page. You can not! You have to scroll to the bottom of the page to make a new container. Then drag it back up to where you want to put one. This wastes time. This is also not consistent.

As mentioned to tech support. I am giving the final websites off to my clients to edit. I have to tell them pages using a page template don't have grab handles, but some do? This is not good.

### Alternate fix if possible
If Elementor doesn't want to fix this (I have no idea why) then they need to add the functionality into the Structure window. 

Currently you can right click and "Add New Container"
This puts a container inside the current container.

If you could right click and "Add New Container Above" or "Add New Container Below" then fine. That would be ok. That would add the functionally back in.

### Conclusion
It is extremely frustrating using page templates. The only way to have a consistent editing experience is to not use them. This is giving up a bunch of functionality and not using Elementors full potential. 

Saying this doesn't break anything is not a good answer. 
This needs to be fixed asap. Please add this to the roadmap and fix. Or at least add my suggestion to the Strutucre window.

Thank you.
